### DISCUSSION TITLE
Add images compatibility in ImageLoaderProps

### DISCUSSION THREAD

**Discussion by EliseiNicolae** [vercel/next.js#43578] [Ideas]
### Describe the feature you'd like to request

I think will be awesome if we'll have 2 booleans that will return if webp / avif are supported by the browser.

isWebPCompatible: boolean
isAVIFCompatible: boolean

### Describe the solution you'd like

I have the following Image example with a loader:

```jsx
<Image
  src={item?.image.source?.url}
  alt={item?.image.alt}
  width={item?.image.source?.width}
  height={item?.image.source?.height}
  quality={90}
  loader={contentfulLoader}
/>
```

What I would expected to be is:

```jsx
const contentfulLoader = ({ src, width, quality, isWebPCompatible, isAVIFCompatible }: any) => {
  if (isAVIFCompatible) {
   return `${src}?w=${width}&q=${quality || 75}&fm=avif`;
  } else if (isWebPCompatible) {
   return `${src}?w=${width}&q=${quality || 75}&fm=webp`;
  }
  return `${src}?w=${width}&q=${quality || 75}&fm=jpg`;
};

export default contentfulLoader;
```

### Describe alternatives you've considered

The second method is by putting the following line on next.config.js

```jsx
  images: {
    formats: ["image/avif", "image/webp", "image/jpg"],
  },
```

And depending on the array order and if it is compatible to return the first compatible format.

```jsx
compatibleFormat: 'avif' | 'webp' | 'jpg';
```

```jsx
const contentfulLoader = ({ src, width, quality, compatibleFormat }: any) => {
  return `${src}?w=${width}&q=${quality || 75}&fm=${compatibleFormat}`;
};

export default contentfulLoader;
```

**Comment by cat-michael-gwynne** [comment]
I'm using lh3.googleusercontent and those URLs accept a load of image options: https://github.com/votemike/lh3-googleusercontent-url-builder/blob/main/index.d.ts
I would love to use next/image but instead have to use a custom made `<picture>` component to support both webp and jpg.
